### DISCUSSION TITLE
HA database (i.e. MySQL). How do you do it?

### DISCUSSION THREAD

**Discussion by MrSnoozles** [coollabsio/coolify#4159] [Q&A]
Hi,

does Coolify already support having a failover if the server where the database is hosted goes down? How do you do it? 
Is there maybe even a way to have databases scaling horizontally? 

Would appreciate if anyone got any insights on this.

**Comment by Winne4r** [comment]
It's not about Coolify. 

To achieve HA, you need to deploy the MySQL Router image and add some configuration. However, HA requires multiple real nodes distributed across different datacenters/locations/providers. You can get more info about it here: https://dev.mysql.com/doc/mysql-shell/8.0/en/mysql-innodb-cluster.html

**Reply by MrSnoozles** [reply]
Cool, thanks for your reply. MariaDB Galera Cluster is an option too: https://mariadb.com/kb/en/what-is-mariadb-galera-cluster/
I was hoping there's already something built-in or someone who has more experience than me has already implemented this with Coolify. Would be very interested to know more and read about how well it worked.
